utopia-remix/app/services/fgaService.server.ts

```diff
@@ -35,6 +35,28 @@ async function checkUserProjectPermission(
   return allowed ?? false
 }
 
+export async function getAllPermissions(
+  projectId: string,
+  userId: string,
+): Promise<Record<FgaUserProjectPermission, boolean>> {
+  const { relations } = await fgaClient.listRelations({
+    user: `user:${userId}`,
+    object: `project:${projectId}`,
+    relations: fgaUserProjectPermission as unknown as string[],
+  })
+  return fgaUserProjectPermission.reduce((acc, permission, index) => {
+    acc[permission] = relations.includes(permission)
+    return acc
+  }, {} as Record<FgaUserProjectPermission, boolean>)
+}
+
+export function getPermissionsOverride(value: boolean): Record<FgaUserProjectPermission, boolean> {
+  return fgaUserProjectPermission.reduce((acc, permission) => {
```

Comment:
In the editor source, we have a function that does this kind of construction called `mapArrayToDictionary` BTW.